hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OzoneClientKeyGenerator.java

```diff
@@ -94,6 +113,15 @@ public Void call() throws Exception {
     contentGenerator = new ContentGenerator(keySize, bufferSize);
     metadata = new HashMap<>();
 
+    if (spec.commandLine().getParseResult().hasMatchedOption("--factor")) {
+      replicationConfig = ReplicationConfig
+          .fromTypeAndFactor(ReplicationType.RATIS, factor);
+    } else {
+      replicationConfig = OzoneClientUtils
+          .validateAndGetClientReplicationConfig(replicationType, replication,
+              ozoneConfiguration);
+    }
```

Comment:
Something is wrong around option handling.  Setting only `replication`, but neither `factor` nor `type`, a regular RATIS/3 key was created:

```
$ ozone freon ockg -n1 -t1 --replication 'rs-3-2-1024k' -p replication_rs-3-2-1024k
...
Successful executions: 1

$ ozone sh key info /vol1/bucket1/replication_rs-3-2-1024k/0
...
  "replicationConfig" : {
    "replicationFactor" : "THREE",
    "requiredNodes" : 3,
    "replicationType" : "RATIS"
  },
...
```

Reply:
It will be meaningless to only setting `replication` or `type` only. 
And it is possible to set `--type EC` with `--replication THREE` together.

| type/replication | THREE | rs-3-2-1024k |
| --- | --- | --- |
| RATIS | OK | Bad |
| EC | Bad | OK |

Should we bind this two options together like `RATIS/THREE` or `EC/rs-3-2-1024k` instead?